### PR TITLE
Move the Checkout -> Account tests from puppeteer to Playwright

### LINKED ISSUE #11893: E2E tests migration: shopper -> account
Migrate E2E tests from `tests/e2e-jest/specs/shopper/cart-checkout/account.test.js` to Playwright.

### PR DESCRIPTION
**Pull request by opr**
## What

Moves the Checkout -> Account tests from puppeteer to Playwright

Fixes #11893

## Why

To move away from Puppeteer.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Run tests, ensure CI passes.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

File: tests/e2e/playwright.config.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import path from 'path';
 
 import { fileURLToPath } from 'url';
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require( 'dotenv' ).config();
```
Comment (opr) [inline tests/e2e/playwright.config.ts:11]: Added so the `WORDPRESS_BASE_URL` env variable is available to tests

File: tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		await requestUtils.rest( {
+			method: 'PUT',
+			path: 'wc/v3/settings/account/woocommerce_enable_guest_checkout',
+			data: { value: 'yes' },
+		} );
+		await requestUtils.rest( {
+			method: 'PUT',
+			path: 'wc/v3/settings/account/woocommerce_enable_checkout_login_reminder',
+			data: { value: 'yes' },
+		} );
```
Comment (tarunvijwani) [inline tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts:40]: Do you think we should move path strings to constants for code readability?

```ts
const GUEST_CHECKOUT_SETTING = 'wc/v3/settings/account/woocommerce_enable_guest_checkout';
const LOGIN_REMINDER_SETTING = 'wc/v3/settings/account/woocommerce_enable_checkout_login_reminder';
const SIGNUP_LOGIN_CHECKOUT_SETTING = 'wc/v3/settings/account/woocommerce_enable_signup_and_login_from_checkout';
```

I have a suggestion. Since we are utilizing the `await requestUtils.rest(` function to configure various settings, do you think it would be better to move it to a separate file so that it can be used easily across multiple test files? We can call that function like this.
```ts
	await configureSettings(requestUtils, GUEST_CHECKOUT_SETTING, 'yes');
	await configureSettings(requestUtils, LOGIN_REMINDER_SETTING, 'yes');
	await configureSettings(requestUtils, SIGNUP_LOGIN_CHECKOUT_SETTING, 'yes');

```

**Review by tarunvijwani — APPROVED** [review]
LGTM! I have left some non-blocking comments.
